src/server/cluster/cluster_family.cc

```diff
@@ -606,15 +593,52 @@ void ClusterFamily::DflyClusterStartSlotMigration(CmdArgList args, ConnectionCon
     slots.emplace_back(SlotRange{slot_start, slot_end});
   } while (parser.HasNext());
 
+  SinkReplyBuilder* rb = cntx->reply_builder();
   if (auto err = parser.Error(); err)
-    return (*cntx)->SendError(err->MakeReply());
+    return rb->SendError(err->MakeReply());
 
-  ClusterSlotMigration node(std::string(host_ip), port, slots);
-  node.Start(cntx);
+  auto* node = AddMigration(std::string(host_ip), port, std::move(slots));
```

Comment:
lets add a check that there is not other migration job from the same host,port pair before adding a new migration

Reply:
I think to make such a check on the source side, because it;s limitation for the source not for target

Reply:
I don't yet have functionality on the source to implement status. Regarding source limitations, I mean that the source can not provide parallel migration, so if we try to do such a migration (it doesn't matter whether it is from 2 nodes or one), the source returns an error.

Reply:
from the doc: "A DF server can run several migration processes from different source hosts at the same time, but only one migration process per source host"
For source, I think creating a separate PR, after I implement the next step of migration, because don't understand how it should look like for now

Reply:
yes from what I remember about this line that we limit the server to run one migration process per source host meaning I cant run several migrations from the same source and target hosts